library/swift/StreamIntel.swift

```diff
@@ -4,14 +4,14 @@ import Foundation
 /// Exposes internal HTTP stream metrics, context, and other details.
 @objcMembers
 public final class StreamIntel: NSObject, Error {
-  // An internal identifier for the stream.
-  public let streamId: UInt64
-  // An internal identifier for the connection carrying the stream.
-  public let connectionId: UInt64
-  // The number of internal attempts to carry out a request/operation.
-  public let attemptCount: UInt64
+  // An internal identifier for the stream. -1 if not set.
+  public let streamId: Int64
```

Comment:
Here, I would make connectionId and streamId optional (so we don't expose our sentinel - which is why I still slightly prefer the bitmask since it would purely be a core/bridge mechanism).

Reply:
The sentinel is already exposed, right? Is your suggestion to now hide it by changing this class? I wonder if that would require changes in the consuming code in our private analytics filter?